Server/src/controllers/DimensionsController.ts

```diff
@@ -0,0 +1,104 @@
+import { Request, Response } from 'express';
+
+import { DimensionService } from '../services/DimensionService';
+import { IDimensionModel } from '../models/interfaces/IDimension'
+import { BadRequest } from '@tsed/exceptions';
+
+/**
+ * This controller is responsible for verifying the user request has correct parameters input.
+ * After request is verified, the appropriate service will be called
+ */
+export class DimensionsController {
+  private dimensionService: DimensionService;
+  private invalidResponse: boolean;
+
+  constructor() {
+    this.dimensionService = new DimensionService();
+  }
+
+  async createDimension(request: Request, response: Response): Promise<Response> {
+    this.invalidResponse = this.validateCreateDimensionRequest(request);
+    if (this.invalidResponse) {
+      return response.status(400).json("Request is invalid. Missing attributes")
+    } else {
+      try {
+        let requestParams: any = { ...request.body };
+        let dimensionInfo: IDimensionModel = requestParams;
+        let requestResponse: any = await this.dimensionService.processAddDimension(dimensionInfo);
+        return response.status(requestResponse.statusCode).json(requestResponse.message);
+      } catch (error) {
+        this.handleError(response, error);
+      }
+    }
+  }
+
+  private validateCreateDimensionRequest(request: Request): boolean {
+    return request.body.name === null || request.body.units[0].name === null;
```

Comment:
we really should do this with Yup. with Yup we can even test if base Unit Id is one of the existing incoming units

Reply:
not for now of course, but for next pr research it

Reply:
Yea maybe if FE adds it, I'll reuse it

Reply:
This might be a good thing to do in the future, as Yup allows for a more expansive verification, but for the time being this is functional.